controllers/clustergroupupgrade_controller.go

```diff
@@ -214,6 +214,12 @@ func (r *ClusterGroupUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.
 				}
 
 				if allManagedPoliciesExist {
+					// Trigger the before enable actions now since they should run even if all the clusters are already compliant.
+					err = r.takeActionsBeforeEnable(ctx, clusterGroupUpgrade)
```

Comment:
The name of this field is misleading. It should have been something like `actionsOnEnable` or `actionsAfterEnable`, e.g. the ztp-running label was meant to show that policy remediation for ztp is in progress. So we can't do it here.

Reply:
Now that I think about it, the `plan == nil` check should have been done later anyways. The blocking CR checks should still be done even if there is nothing to do for this CGU.